### PR TITLE
feat: Display a modal before detaching a doc

### PR DESCRIPTION
**Pull request by Crash--**
Display a modal before unlinking a document. 

Feel free to edit the english :D

### CODE REVIEW

File: packages/cozy-procedures/src/components/documents/DocumentHolder.jsx [hunk 1]
```diff
@@ -1,33 +1,73 @@
...
+    return (
+      <>
+        {isModalOpened && (
+          <Modal
+            primaryText={t('documents.unlink.unlink')}
+            primaryAction={() => unlinkDocument({ document, documentId })}
+            secondaryText={t('cancel')}
+            secondaryAction={() => this.setState({ isModalOpened: false })}
+            dismissAction={() => this.setState({ isModalOpened: false })}
+            overflowHidden={true}
```
Comment (y-lohse) [inline packages/cozy-procedures/src/components/documents/DocumentHolder.jsx]: Why do we need this prop?

File: packages/cozy-procedures/src/components/documents/DocumentHolder.jsx [hunk 2]
```diff
@@ -1,33 +1,73 @@
...
-    </Chip>
-  )
+class DocumentHolder extends Component {
+  state = {
+    isModalOpened: false
+  }
+
+  render() {
+    const { document, unlinkDocument, documentId, t } = this.props
+    const { isModalOpened } = this.state
```
Comment (cedricmessiant) [inline packages/cozy-procedures/src/components/documents/DocumentHolder.jsx]: maybe be more precise about the modal : `isUnlinkConfirmationModalOpened` is a bit long though ;-)

File: packages/cozy-procedures/src/components/documents/DocumentHolder.jsx [hunk 3]
```diff
@@ -1,33 +1,73 @@
...
+    const splittedName = CozyFile.splitFilename(document)
+    return (
+      <>
+        {isModalOpened && (
+          <Modal
+            primaryText={t('documents.unlink.unlink')}
+            primaryAction={() => unlinkDocument({ document, documentId })}
+            secondaryText={t('cancel')}
+            secondaryAction={() => this.setState({ isModalOpened: false })}
+            dismissAction={() => this.setState({ isModalOpened: false })}
```
Comment (y-lohse) [inline packages/cozy-procedures/src/components/documents/DocumentHolder.jsx]: I think one of dismiss or secondaryAction is optional, but no big deal.

File: packages/cozy-procedures/src/locales/en.json [hunk 4]
```diff
@@ -128,7 +129,12 @@
       "conflict_error": "A file with the same name is already uploaded. Please rename your file.",
       "error": "An error occured during upload"
     },
-
+    "unlink": {
+      "title": "Detach from your application?",
+      "subtitle": "You'll detach this file from your application.",
+      "text": "The file will not be deleted from your Document",
```
Comment (cedricmessiant) [inline packages/cozy-procedures/src/locales/en.json]:
```suggestion
      "text": "The file will not be deleted from your documents",
```

or maybe "from Drive" and override by "from Documents" in the customer's project

File: packages/cozy-procedures/src/locales/fr.json [hunk 5]
```diff
@@ -128,7 +129,12 @@
       "conflict_error": "Un fichier avec le même nom est déjà présent. Veuillez renommer votre fichier avant de recommencer.",
       "error": "Une erreur est survenue durant l'upload du fichier."
     },
-
+    "unlink": {
+      "title": "Détacher du dossier ? ",
```
Comment (cedricmessiant) [inline packages/cozy-procedures/src/locales/fr.json]:
```suggestion
      "title": "Détacher du dossier ?",
```

File: packages/cozy-procedures/src/components/documents/DocumentHolder.jsx (hunk 1 above)
Comment (Crash--) [reply packages/cozy-procedures/src/components/documents/DocumentHolder.jsx]: I have copy pasted from the styleguidish. Removing it

**Pull request merged**